### PR TITLE
Added BoxLineGeometry

### PR DESCRIPTION
**Pull request by mrdoob**
Using `wireframe: true` has always been a hack which produced ~20x more lines.

### CODE REVIEW

**Comment by mrdoob** [comment]
Would be nice to also create `VolumeLineGeometry` and use it to visualise [webgl_marchingcubes](https://threejs.org/examples/webgl_marchingcubes.html).

Up for it @Mugen87?

File: examples/js/geometries/BoxLineGeometry.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	var segmentWidth = width / widthSegments;
+	var segmentHeight = height / heightSegments;
+	var segmentDepth = depth / depthSegments;
+
+	var vertices = [];
+
+	var x = - widthHalf, y = - heightHalf, z = - depthHalf;
+
+	for ( var i = 0; i < widthSegments; i ++ ) {
```
Comment (WestLangley) [inline examples/js/geometries/BoxLineGeometry.js]:
```javascript
for ( var i = 0; i <= widthSegments; i ++ ) { // all 3 occurrences
```

Comment (WestLangley) [reply examples/js/geometries/BoxLineGeometry.js]: I see it is fixed now.

Comment (mrdoob) [reply examples/js/geometries/BoxLineGeometry.js]: Yeah 😅

**Comment by Mugen87** [comment]
> Up for it @Mugen87?

I'll put it on my todo list 😇

**Comment by Mugen87** [comment]
@mrdoob I have a question about `VolumeLineGeometry` 😊. `MarchingCubes` creates dynamically geometric data which is a different approach like in this PR where the geometry is static and does not change over time. What does this mean for `VolumeLineGeometry`? Do we need an `update()` method in order to refresh `VolumeLineGeometry`'s data from the current position values of `MarchingCubes` each frame?

**Pull request merged**